tools/pom.xml

```diff
@@ -135,6 +135,19 @@
                 <groupId>net.alchim31.maven</groupId>
                 <artifactId>scala-maven-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-source-plugin</artifactId>
```

Comment:
Need to add plugin `maven-source-plugin` to generate tools-sources.jar.

Though the parent pom.xml [Line590 ](https://github.com/NVIDIA/spark-rapids/blob/branch-21.10/pom.xml#L590)defined this plugin, seems tools module can not inherit it?

Reply:
To generate tools-javadoc.jar, also we need to remove `-Dmaven.javadoc.skip=true` from https://gitlab-master.nvidia.com/tgraves/blossom-jenkins/-/blob/deployrelease/spark-rapids/Jenkinsfile.release#L96

Reply:
Though the parent pom.xml [Line590 ](https://github.com/NVIDIA/spark-rapids/blob/branch-21.10/pom.xml#L590)defined this plugin, seems tools module can not inherit it? Not know why it (Line590 ) does not work for tools module, but it works for other modules like shims or sql-plugin.

```
[2021-09-26T01:51:02.232Z] [INFO] --- maven-source-plugin:3.0.0:jar-no-fork (attach-source) @ rapids-4-spark-sql_2.12 ---
[2021-09-26T01:51:02.232Z] [INFO] Building jar: /home/jenkins/agent/workspace/jenkins-rapids-release-63/sql-plugin/target/rapids-4-spark-sql_2.12-21.10.0-sources.jar

```

Reply:
these seem to generate fine locally for me without adding these
> mvn -pl tools -Psource-javadoc,snpshots -Dbuildver=311  install -DskipTests

both javadoc and sources jar are generated.  I'll try running another test